frontend/src/stores/filterState/FilterState.ts

```diff
@@ -36,14 +36,167 @@ export interface FilterLineState {
+  public setFilterLineActive(
+    filterLineType: string,
+    entity: 'node' | 'edge'
+  ): void {
+    const line: FilterLineState | undefined = this.getLineFromFilterState(
+      filterLineType,
+      entity
+    );
+
+    if (line) {
+      line.isActive = true;
+    }
+  }
+
+  /**
+   * Replaces {@link FilterPropertyState}s of the specified filter line.
+   * @param statesToReplace - the {@link FilterPropertyState}s to be replaced
+   * @param filterLineType - filter line which the {@link FilterPropertyState} is added to
+   * @param entity - specifies whether this entity is a node or an edge
+   */
+  public replaceFilterPropertyStates(
+    statesToReplace: FilterPropertyState[],
+    filterLineType: string,
+    entity: 'node' | 'edge'
+  ): void {
+    const lineState: FilterLineState | undefined = this.getLineFromFilterState(
+      filterLineType,
+      entity
+    );
+
+    if (lineState) {
+      lineState.propertyFilters = statesToReplace;
+    }
+  }
+
+  /**
+   * Gets {@link FilterPropertyState}s of the specified filter line.
+   * @param filterLineType - filter line which the {@link FilterPropertyState} is added to
+   * @param entity - specifies whether this entity is a node or an edge
+   */
+  public getFilterPropertyStates(
+    filterLineType: string,
+    entity: 'node' | 'edge'
+  ): FilterPropertyState[] | undefined {
+    const lineState: FilterLineState | undefined = this.getLineFromFilterState(
+      filterLineType,
+      entity
+    );
+
+    return lineState?.propertyFilters;
+  }
+
+  /**
+   * Gets the values of a {@link FilterPropertyState} in the specified filter line if they exist.
+   * Otherwise undefined is returned.
+   * @param filterLineType - filter line which is searched for
+   * @param filterPropertyName - name of the {@link FilterPropertyState}
+   * @param entity - specifies whether this entity is a node or an edge
+   */
+  public getPropertyStateValues(
+    filterLineType: string,
+    filterPropertyName: string,
+    entity: 'node' | 'edge'
+  ): string[] | undefined {
+    const propertyState: FilterPropertyState | undefined =
+      this.searchForPropertyState(filterLineType, filterPropertyName, entity);
+
+    return propertyState?.values;
+  }
+
+  private searchForPropertyState(
+    filterLineType: string,
+    filterPropertyName: string,
+    entity: 'node' | 'edge'
+  ) {
+    const line: FilterLineState | undefined = this.getLineFromFilterState(
+      filterLineType,
+      entity
+    );
+
+    if (line) {
+      for (const propertyFilters of line.propertyFilters) {
+        if (propertyFilters.name === filterPropertyName) {
+          return propertyFilters;
+        }
+      }
+    }
+
+    return undefined;
+  }
+
+  private getLineFromFilterState(
+    filterLineType: string,
+    entity: 'node' | 'edge'
+  ): FilterLineState | undefined {
+    const entityLineStates = entity === 'node' ? this.nodes : this.edges;
+
+    for (const entityLine of entityLineStates) {
```

Comment:
This can be simplified to `entityLineStates.find((entityLine ) => entityLine.type === filterLineType);`
See: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find  
This also correctly returns `undefined` if the value cannot be found.